src/core/portal/portal.ts

```diff
@@ -60,21 +67,28 @@ export abstract class Portal<T> {
 /**
  * A `ComponentPortal` is a portal that instantiates some Component upon attachment.
  */
-export class ComponentPortal extends Portal<ComponentRef<any>> {
+export class ComponentPortal<T> extends Portal<ComponentRef<T>> {
   /** The type of the component that will be instantiated for attachment. */
-  public component: Type;
+  component: ComponentType<T>;
 
   /**
    * [Optional] Where the attached component should live in Angular's *logical* component tree.
    * This is different from where the component *renders*, which is determined by the PortalHost.
    * The origin necessary when the host is outside of the Angular application context.
    */
-  public viewContainerRef: ViewContainerRef;
+  viewContainerRef: ViewContainerRef;
+
+  /** [Optional] Injector used for the instantiation of the component. */
+  injector: Injector;
```

Comment:
That syntax is only for an `interface` (saying that implementing it is optional). When a _value_ is optional, it just means that the value can be `null` or `undefined`